safe/gui/tools/multi_buffer_dialog.py

```diff
@@ -0,0 +1,229 @@
+        self.setupUi(self)
+        self.setWindowTitle(self.tr('InaSAFE Multi Buffer Tool'))
+
+        self.output_layer = None
+
+        # set icon
+        self.add_class_button.setIcon(
+            QIcon(resources_path('img', 'icons', 'add.svg')))
+        self.remove_class_button.setIcon(
+            QIcon(resources_path('img', 'icons', 'remove.svg')))
+
+        # prepare dialog initialisation
+        self.layer.setFilters(QgsMapLayerProxyModel.VectorLayer)
+        self.directory_button.setEnabled(False)
+        self.add_class_button.setEnabled(False)
+        self.button_box.button(QtGui.QDialogButtonBox.Ok).setEnabled(False)
+
+        # set signal
+        self.layer.layerChanged.connect(self.get_output_from_input)
+        self.layer.layerChanged.connect(self.directory_button_status)
+        self.layer.layerChanged.connect(self.ok_button_status)
+        self.output_form.textChanged.connect(self.ok_button_status)
+        self.directory_button.clicked.connect(
+            self.on_directory_button_tool_clicked)
+        self.radius_form.valueChanged.connect(self.add_class_button_status)
+        self.class_form.textChanged.connect(self.add_class_button_status)
+        self.add_class_button.clicked.connect(
+            self.populate_hazard_classification)
+        self.add_class_button.clicked.connect(self.ok_button_status)
+        self.remove_class_button.clicked.connect(
+            self.remove_selected_classification)
+        self.remove_class_button.clicked.connect(self.ok_button_status)
+
+        # Set up things for context help
+        self.help_button = self.button_box.button(QtGui.QDialogButtonBox.Help)
+        # Allow toggling the help button
+        self.help_button.setCheckable(True)
+        self.help_button.toggled.connect(self.help_toggled)
+        self.main_stacked_widget.setCurrentIndex(1)
+
+        # Fix for issue 1699 - cancel button does noth  ing
+        cancel_button = self.button_box.button(QtGui.QDialogButtonBox.Cancel)
+        cancel_button.clicked.connect(self.reject)
+        # Fix ends
+        ok_button = self.button_box.button(QtGui.QDialogButtonBox.Ok)
+        ok_button.clicked.connect(self.accept)
+
+    def accept(self):
+        """Process the layer for multi buffering and generate a new layer.
+
+        .. note:: This is called on OK click.
+        """
+        # set parameter from dialog
+        input_layer = self.layer.currentLayer()
+        output_path = self.output_form.text()
+        file_name = os.path.splitext(output_path)[0].split('/')[-1]
+        radius = self.get_classification()
+        # monkey patch keywords so layer works on multi buffering function
+        input_layer.keywords = {'inasafe_fields': {}}
+
+        # run multi buffering
+        self.output_layer = multi_buffering(input_layer, radius)
+
+        # save output layer to datastore
+        data_store = Folder(os.path.dirname(output_path))
+        if self.output_form.text().endswith('.shp'):
```

Comment:
I though we were going to remove shp ?

If not, let's add a line : `data_store.default_vector_format = 'shp'` jsut in case we change the default behaviour soon.

* data_store.add_layer(self.output_layer, file_name) : remove one indent and use this line only one time ;-) (l104 and 107)